src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,173 @@
-                    .withName(logLocation)
-                    .setTarget(ConsoleAppender.Target.SYSTEM_ERR)
-                    .withLayout(layout)
-                    .build();
-
-                ca.start();
-                config.addAppender(ca);
-                ctx.getRootLogger().addAppender(config.getAppender(ca.getName()));
+            @Override
+            public synchronized String format(LogRecord lr) {
+                return String.format(format,
+                    dateFormatter.format(new Date()).toString(),
+                    LogLevel.fromJulLevel(lr.getLevel()).toString(),
+                    simpleClassName(lr.getSourceClassName()),
+                    lr.getMessage()
+                );
+            }
+        };
+
+        // prepare the different handlers
+        // --
+
+        ConsoleHandler stdoutHandler = null;
+        ConsoleHandler stderrHandler = null;
+        FileHandler fileHandler = null;
+
+        // the logLocation isn't always containing a file, it is sometimes
+        // referring to a standard output. We want to create a FileHandler only
+        // if the logLocation is a file on the FS.
+        if (logLocation != null && logLocation.length() > 0) {
+            if (!isStdOut(logLocation) && !isStdErr(logLocation)) {
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, MAX_FILE_SIZE, FILE_COUNT);
+                    fileHandler.setFormatter(formatter);
+                } catch (Exception e) {
+                    fileHandler = null;
+                    log.error("can't open the file handler:", e);
+                }
+            } else if (isStdErr(logLocation)) {
+                // console handler sending on stderr
+                // note that ConsoleHandler is sending on System.err
+                stderrHandler = new ConsoleHandler();
+                stderrHandler.setFormatter(formatter);
+                stderrHandler.setLevel(level.toJulLevel());
             }
         }
 
-        // replace default appender with the correct layout
-        LoggerConfig loggerConfig = config.getLoggerConfig(LogManager.ROOT_LOGGER_NAME);
-        loggerConfig.removeAppender(target);
+        // always have a console handler sending the logs to stdout
+        stdoutHandler = new StdoutConsoleHandler();
+        stdoutHandler.setFormatter(formatter);
+        stdoutHandler.setLevel(level.toJulLevel());
 
-        Appender appender = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(target)
-                    .withLayout(layout)
-                    .build();
-        appender.start();
+        // should configure the root logger
+        // ---
 
-        loggerConfig.addAppender(appender, null, null);
-        loggerConfig.setLevel(level);
+        Logger logger = Logger.getLogger("");
```

Comment:
I think we need to keep a reference to this logger (maybe at the level of the `CustomLogger` class) otherwise the logger might get garbage-collected if I read https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#getLogger(java.lang.String) correctly (in which case I think it'd get completely reset after getting GC'ed).

Since `""` returns the root logger, I may have missed some additional documentation that explains that the root logger never gets GC'ed maybe 🤔

Reply:
Looks like root logger doesn't need a strong ref afaict: https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/logging/LogManager.java#L464-L465

Reply:
Good find! Should this be regarded as an implementation detail that may be different in other JVM implementations? (I'd feel safer with a strong ref to the root logger in a class attribute, but I may be a bit paranoid)

Reply:
I think this isn't an implementation detail since the root logger holds the default handlers. With that said, I don't think there would be harm in using a strong ref just in case.

Reply:
I don't think we need a strong reference in our code.